lib/conjur/feature_flags/features.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module Conjur
+  module FeatureFlags
+    # Provides feature flagging support
+    class Features
+      # Instantiates a new Conjur::FeatureFlags::Features object
+      #
+      # @param logger [Logger] the logger used for output
+      # @param environment_settings [Hash] the environment variable has (typically ENV)
+      # @param feature_flags [Array<Symbol>] the list of valid feature flags
+      #
+      # @return [Conjur::FeatureFlags::Features]
+      def initialize(logger:, environment_settings:, feature_flags: [])
+        @logger = logger
+        @enabled_features = merge_with_environment_variable_flags(
+          features: feature_flags,
+          env: environment_settings
+        )
+      end
+
+      # Check to see if a feature flag has been enabled
+      #
+      # @param feature_name [Symbol] the feature flag we want to check
+      #
+      # @return [Boolean]
+      def enabled?(feature_name)
+        @logger.debug("Conjur::FeatureFlags::Features#enabled? - feature flag name: '#{feature_name.inspect}'")
+        raise InvalidArguement unless feature_name.is_a?(Symbol)
+
+        symbolized_feature_name = feature_name.downcase
```

Comment:
Not a blocker, but I'd remove the downcase here for the same reason we removed the other courtesy code.  Clients should be using the downcased symbols, and it's an error if they don't.